grapher/text/MarkdownTextWrap.tsx

```diff
@@ -0,0 +1,527 @@
+    fontSize: number
+    fontWeight?: number
+} & (MarkdownTextWrapHTMLProps | MarkdownTextWrapSVGProps)
+
+export class MarkdownTextWrap extends React.Component {
+    props: MarkdownTextWrapProps
+    constructor(props: MarkdownTextWrapProps) {
+        super(props)
+        this.props = props
+    }
+
+    @computed get maxWidth(): number {
+        return this.props.maxWidth ?? Infinity
+    }
+    @computed get lineHeight(): number {
+        return this.props.lineHeight ?? 1.1
+    }
+    @computed get fontSize(): number {
+        return this.props.fontSize ?? 1
+    }
+    @computed get fontWeight(): number | undefined {
+        return this.props.fontWeight
+    }
+    @computed get text(): string {
+        return this.props.text
+    }
+    @computed get ast(): MarkdownRoot["children"] {
+        const result = mdParser.markdown.parse(this.props.text)
+        if (result.status) {
+            return result.value.children
+        }
+        return []
+    }
+
+    @computed get lines(): IRToken[][] {
+        const tokens = parsimmonToTextTokens(this.ast)
+        return splitIntoLines(tokens, this.maxWidth)
+    }
+
+    // @computed get height(): number {
+    //     if (this.lines.length === 0) return 0
+
+    //     return (
+    //         this.lines.reduce((total, line) => total + line.height, 0) +
+    //         this.lineHeight * (this.lines.length - 1)
+    //     )
+    // }
+
+    // @computed get width(): number {
+    //     return max(this.lines.map((l) => l.width)) ?? 0
+    // }
+
+    @computed get htmlStyle(): any {
+        const { fontSize, fontWeight, lineHeight } = this
+        return {
+            fontSize: fontSize.toFixed(2) + "px",
+            fontWeight: fontWeight,
+            lineHeight: lineHeight,
+            overflowY: "visible",
+        }
+    }
+
+    render(): JSX.Element | null {
+        const { props, lines, fontSize, fontWeight, lineHeight } = this
+
+        // if (lines.length === 0) return null
+
+        if (props.isSVG) {
+            const yOffset = 1
+            // const yOffset = props.y + lines[0].height - lines[0].height * 0.2
+            return (
+                <text
+                    fontSize={fontSize.toFixed(2)}
+                    fontWeight={fontWeight}
+                    x={props.x.toFixed(1)}
+                    y={yOffset.toFixed(1)}
+                    {...props.svgOptions}
+                >
+                    {lines.map((line) =>
+                        line.map((token, i) => token.toSVG(i))
+                    )}
+                </text>
+            )
+        }
+        return (
+            <span>
+                {lines.map((line, i) => (
+                    <div key={i}>
+                        {line.length ? (
+                            line.map((token, i) => token.toHTML(i))
+                        ) : (
+                            <br />
+                        )}
+                    </div>
+                ))}
+            </span>
```

Comment:
In `TextWrap`, we use`<span>`: https://github.com/owid/owid-grapher/blob/31fc73b21799963e97f908b72b601a85f39979aa/grapher/text/TextWrap.tsx#L166-L190

I'm not sure if it makes a difference, but if we match the output identically, we can check the diffs in `owid-static` for any regressions.

I made them `<div>`s for no good reason, it was just an easy way to implement lines.

Reply:
On second thought maybe not: no charts are rendered as HTML in the baked files, but might still be a good idea to match output as closely as possible – up to you!

Reply:
Yup I'll get these details right once I've got the high level work in the right place, this was just a temporary commit to show off the overall strategy 👍